packages/hardhat-core/src/internal/cli/project-creation.ts

```diff
@@ -142,6 +142,44 @@ async function printWelcomeMessage() {
   );
 }
 
+async function checkForDuplicates(
+  dest: string,
+  projectType: SampleProjectTypeCreationAction
+): Promise<void> {
+  const { intersection, union } = await import("lodash");
+
+  const packageRoot = getPackageRoot();
+
+  const srcPath = path.join(packageRoot, "sample-projects");
+  const destFiles = fsExtra.readdirSync(dest);
+  let srcFiles: string[] = fsExtra.readdirSync(path.join(srcPath, "basic"));
+
+  switch (projectType) {
+    case Action.CREATE_ADVANCED_SAMPLE_PROJECT_ACTION:
+      srcFiles = union(
+        srcFiles,
+        fsExtra.readdirSync(path.join(srcPath, "advanced"))
+      );
+      break;
+    case Action.CREATE_ADVANCED_TYPESCRIPT_SAMPLE_PROJECT_ACTION:
+      srcFiles = union(
+        srcFiles,
+        fsExtra.readdirSync(path.join(srcPath, "advanced")),
+        fsExtra.readdirSync(path.join(srcPath, "advanced-ts"))
```

Comment:
Won't this warn about unnecessary things? For example, `hardhat.config.js` in a ts project.

I don't remember how the `advanced-ts` project creation works. Is it applied on top of the `advanced` one?

Reply:
I'm ok with this trade-off, my guess is that it won't matter that much in practice.